### PR TITLE
created entry for JavaScript section

### PR DESCRIPTION
**Pull request by Dusch4593**
Created an entry, "Hoisting in JavaScript".

### CODE REVIEW

File: content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+var myName = "Brandon"
+```
+
+In the code block, we are logging `myName` to the console before it is declared and assigned. Doing so will yield `undefined`. While JavaScript's compiler has stored the *declaration* for `myName` in memory, its definition is still not known. Therefore, its value and type is `undefined`. 
+
+During the execution phase, the code will refactor like this: 
+
+```js
+var myName
+console.log(`My name is ${myName}.`) // undefined
```
Comment (sonnynomnom) [inline content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md:36]: I added `// Output: `

File: content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Catalog Content:
+  - "https://www.codecademy.com/learn/learn-javascript"
+  - "https://www.codecademy.com/learn/paths/web-development"
+  - "https://www.codecademy.com/learn/paths/computer-science"
+---
+
+# Hoisting in JavaScript
+
+## Introduction
+**Hoisting** is a process where JavaScript "splits" `var` and `function` declarations from their definition. These declarations are "moved up" to the top of the file above where they were defined and/or used. This means that JavaScript knows about these declarations before the rest of the code is executed.
```
[the remark this answers is not in the document]
Comment (sonnynomnom) [reply content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md:19]: definition to definitions (I think)

File: content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+---
+
+# Hoisting in JavaScript
+
+## Introduction
+**Hoisting** is a process where JavaScript "splits" `var` and `function` declarations from their definition. These declarations are "moved up" to the top of the file above where they were defined and/or used. This means that JavaScript knows about these declarations before the rest of the code is executed.  
+
+
+### Variables
+In JavaScript, variables declared with the `var` keyword are hoist-able.
```
Comment (sonnynomnom) [inline content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md:23]: Let's add a new line after markdown headings.

File: content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+If we tried to log a variable that hasn't been defined yet, we'd get a `ReferenceError`:
+
+```js
+console.log(`My name is ${name}.`) // --> ReferenceError
+var myName = "Brandon"
+```
+
+Now that we've explore hoisting with variables, let's move on to functions.
```
Comment (sonnynomnom) [inline content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md:47]: For Codecademy Docs, we don't really need transition sentences.

**Review by sonnynomnom — COMMENTED** [review]
Already made these changes btw!

File: content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+### Functions
+In addition to variables, *functions* can be hoisted in JavaScript. This is mainly because they are "first-class objects" and they can be stored as variables. The declarations for functions, like the ones for variables, are hoisted during the execution phase: 
+
+```js
+console.log(greetings()) // Hi! My name is Brandon.
+function greetings() {
+    var myName = "Brandon"
```
[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/javascript/concepts/hoisting-in-javascript/hoisting-in-javascript.md:55]: That's good to know. I'll add them going forward!

**Comment by sonnynomnom** [comment]
Just realized this is PR #100 haha @Dusch4593

**Pull request merged**